packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -552,6 +566,13 @@ function createArrayNodeProxy(
 	allowAdditionalProperties: boolean,
 	customTargetObject?: object,
 ): TreeArrayNode {
+	if (customTargetObject !== undefined) {
+		// This causes `JSON.stringify` to give the same output as if the proxy were truly an array.
+		// Otherwise, it will stringify like an object (e.g. `{"0": ..., "1": ...}`) for a `customTargetObject`.
```

Comment:
I think we should mention somewhere why we don't do this approach for non-custom targets.

Reply:
No longer relevant - moved to the class itself.